starknet_py/net/full_node_client.py

```diff
@@ -454,3 +446,53 @@ def _get_call_payload(tx: Union[InvokeFunction, Call]) -> dict:
         "entry_point_selector": convert_to_felt(tx.selector),
         "calldata": [convert_to_felt(i) for i in tx.calldata],
     }
+
+
+def _create_broadcasted_txn(transaction: Union[InvokeFunction, Declare]) -> Dict:
+    common_params = {
+        "type": "INVOKE",
+        "max_fee": hex(transaction.max_fee),
+        "version": hex(transaction.version),
+        "signature": [convert_to_felt(i) for i in transaction.signature],
+        "nonce": convert_to_felt(transaction.nonce)
+        if transaction.nonce is not None
+        else transaction.nonce,
+    }
+
+    if transaction.tx_type == TransactionType.INVOKE_FUNCTION:
+        invoke_specific_params = {
+            "calldata": [convert_to_felt(i) for i in transaction.calldata],
+        }
+        if transaction.version == 0:
+            params_depending_on_version = {
+                "contract_address": convert_to_felt(transaction.contract_address),
+                "entry_point_selector": convert_to_felt(
+                    transaction.entry_point_selector
+                ),
+            }
+        else:
+            params_depending_on_version = {
+                "sender_address": convert_to_felt(transaction.contract_address),
+            }
+
+        return {
+            **common_params,
+            **params_depending_on_version,
+            **invoke_specific_params,
+        }
+
+    if transaction.tx_type == TransactionType.DECLARE:
+        contract_class = transaction.dump()["contract_class"]
+
+        declare_params = {
+            "contract_class": {
+                "program": contract_class["program"],
+                "entry_points_by_type": contract_class["entry_points_by_type"],
+                "abi": contract_class["abi"],
+            },
+            "sender_address": convert_to_felt(transaction.sender_address),
+        }
+
+        return {**common_params, **declare_params}
+
+    return {}
```

Comment:
I don't think we should default to empty dict. If transaction is neither DECLARE nor INVOKE, this method should fail.